CMakeLists.txt

```diff
@@ -88,16 +88,30 @@ if(NOT BUILD_SHARED_LIBS)
       DESTINATION lib/swift_static/$<LOWER_CASE:${CMAKE_SYSTEM_NAME}>)
   endif()
 endif()
+
+set(swift_lib_dir "lib/swift")
+if(NOT BUILD_SHARED_LIBS)
+  set(swift_lib_dir "lib/swift_static")
+endif()
+
 # TODO(compnerd) install as a Framework as that is how swift actually is built
 install(DIRECTORY
           ${CMAKE_CURRENT_BINARY_DIR}/CoreFoundation.framework/Headers/
         DESTINATION
           lib/swift/CoreFoundation
         FILES_MATCHING PATTERN "*.h")
-install(FILES
-          CoreFoundation/Base.subproj/module.map
-        DESTINATION
-          lib/swift/CoreFoundation)
+if(CMAKE_SYSTEM_NAME STREQUAL WASI)
+  install(FILES
+            CoreFoundation/Base.subproj/$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:static/>wasm32-unknown-wasi.modulemap
+          DESTINATION
+            ${swift_lib_dir}/CoreFoundation
```

Comment:
Please keep to use `lib/swift` dir because we already some [build script](https://github.com/swiftwasm/swift/blob/swiftwasm-release/5.3/utils/webassembly/build-foundation.sh#L28) that assumes it is located under `lib/swift`.